core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1245,9 +1250,8 @@ class DAGScheduler(
                     markMapStageJobAsFinished(job, stats)
                   }
                 }
+                submitWaitingChildStages(shuffleStage)
```

Comment:
I don't think we need.
The `shuffleStage` is re-submitted if it is not available, so the `submitWaitingChildStages(shuffleStage)` for the stage will call after it is processed again and becomes available.